keps/sig-auth/4940-psa-block-host-field-in-probes/README.md

```diff
@@ -0,0 +1,370 @@
+
+
+###### Can the feature be disabled once it has been enabled (i.e. can we roll back the enablement)?
+
+No
+
+###### What happens if we reenable the feature if it was previously rolled back?
+
+N/A
+
+###### Are there any tests for feature enablement/disablement?
+
+N/A since there is no feature gate
+
+### Rollout, Upgrade and Rollback Planning
+
+
+###### How can a rollout or rollback fail? Can it impact already running workloads?
+
+N/A
+
+###### What specific metrics should inform a rollback?
+
+N/A
+
+###### Were upgrade and rollback tested? Was the upgrade->downgrade->upgrade path tested?
+
+N/A
+
+###### Is the rollout accompanied by any deprecations and/or removals of features, APIs, fields of API types, flags, etc.?
+
+N/A
+
+### Monitoring Requirements
+
+N/A
+
+###### How can an operator determine if the feature is in use by workloads?
+
+If pods have probes with `.Host` field set and PSA annotation is set on that pod's namespace
+to a version where the new admission has been added, then it means the feature is enabled.
+
+###### How can someone using this feature know that it is working for their instance?
+
+Trying to create a pod with `.Host` field set in the probes will fail
+like this:
+```
+Error from server (Forbidden): error when creating "psa/fail-case-pod.yaml": pods "liveness-http-pass" is forbidden: violates PodSecurity "restricted:latest": probeHost (container "liveness" uses probeHost 135.45.63.4)
+```
+
+###### What are the reasonable SLOs (Service Level Objectives) for the enhancement?
+N/A
+
+###### What are the SLIs (Service Level Indicators) an operator can use to determine the health of the service?
+
+N/A
+
+###### Are there any missing metrics that would be useful to have to improve observability of this feature?
+
+N/A
+
+### Dependencies
+
+None
+
+###### Does this feature depend on any specific services running in the cluster?
+
+No
+
+### Scalability
+
+N/A
+
+###### Will enabling / using this feature result in any new API calls?
+
+No
+
+###### Will enabling / using this feature result in introducing new API types?
+
+No
+
+###### Will enabling / using this feature result in any new calls to the cloud provider?
+
+No
+
+###### Will enabling / using this feature result in increasing size or count of the existing API objects?
+
+No
+
+###### Will enabling / using this feature result in increasing time taken by any operations covered by existing SLIs/SLOs?
+
+No
+
+###### Will enabling / using this feature result in non-negligible increase of resource usage (CPU, RAM, disk, IO, ...) in any components?
+
+No
+
+###### Can enabling / using this feature result in resource exhaustion of some node resources (PIDs, sockets, inodes, etc.)?
+
+No
+
+### Troubleshooting
+
+###### How does this feature react if the API server and/or etcd is unavailable?
+
+N/A
+
+###### What are other known failure modes?
+
+N/A
+
+###### What steps should be taken if SLOs are not being met to determine the problem?
```

Comment:
these need to be filled in with at least references to the PSA docs about the difference between latest and pinned to versions.

Reply:
hmm sorry for the stupid q: "pinned to version" here is the one where this change will be introduced?
let me check a sample past PSA enhancement to see an example.

Reply:
ah I understand now, gotcha, included this info now in latest push